backend/engine/plugins/trivy_sbom/main.py

```diff
@@ -0,0 +1,145 @@
+"""
+import json
+import subprocess
+from engine.plugins.lib import utils
+from engine.plugins.lib.trivy_common.generate_locks import check_package_files
+from engine.plugins.lib.sbom_common.go_installer import go_mod_download
+from engine.plugins.lib.sbom_common.yarn_installer import yarn_install
+from engine.plugins.trivy_sbom.parser import clean_output_application_sbom
+
+logger = utils.setup_logging("trivy_sbom")
+
+NO_RESULTS_TEXT = "no supported file was detected"
+
+
+# Scan the repo at an application level
+def execute_trivy_application_sbom(path: str, include_dev: bool):
+    logger.info(f"Creating SBOM at an application level. Dev-dependencies: {include_dev}")
+    args = ["trivy", "fs", path, "--format", "cyclonedx"]
+    if include_dev:
+        args.append("--include-dev-deps")
+    proc = subprocess.run(args, capture_output=True, check=False)
+    if proc.returncode != 0:
+        logger.warning(proc.stderr.decode("utf-8"))
+        return None
+    if proc.stdout.decode("utf-8") == "null":
+        logger.warning("No response returned. No files to parse.")
+        return None
+    if proc.stdout and NO_RESULTS_TEXT not in proc.stdout.decode("utf-8"):
+        return proc.stdout.decode("utf-8")
+    logger.error(proc.stderr.decode("utf-8"))
+    return None
+
+
+# Scan the images
+def execute_trivy_image_sbom(image: str):
+    proc = subprocess.run(["trivy", "image", image, "--format", "cyclonedx"], capture_output=True, check=False)
+    if proc.returncode != 0:
+        logger.warning(proc.stderr.decode("utf-8"))
+        return None
+    if proc.stdout:
+        return proc.stdout.decode("utf-8")
+    logger.error(proc.stderr.decode("utf-8"))
+    return None
+
+
+def process_docker_images(images: list):
+    """
+    Pulls a list of image information, scans the successful ones, and returns the outputs.
+    example list item:
+    """
+    outputs = []
+    parsed = []
+    for image in images:
+        if not image.get("status"):
+            continue
+        try:
+            output = convert_output(execute_trivy_image_sbom(image["tag-id"]))
+            if not output:
+                logger.warning(
+                    "Image from Dockerfile %s could not be scanned or the results converted to JSON",
+                    image["dockerfile"],
+                )
+            else:
+                outputs.append(output)
+                parsed.extend(clean_output_application_sbom(output))
+        except Exception as e:
+            logger.warning("Issue scanning image: %s", e)
+
+    logger.info("Successfully scanned %d images", len(outputs))
+    return outputs, parsed
+
+
+def build_scan_parse_images(images) -> list:
+    results = []
+    parsed = []
+    logger.info("Dockerfiles found: %d", images["dockerfile_count"])
+    outputs, parsed_image = process_docker_images(images["results"])
+    results.extend(outputs)
+    parsed.extend(parsed_image)
+    return results, parsed
+
+
+def convert_output(output_str: str):
```

Comment:
yea I can do that, it's just that other places in the code base, convert_output was used for performing this functionality so thought to keep it the same.  If I do change it, It would be better to change that throughout the code base or maybe just make this a shared function and just have every file just import this in. Thoughts?

Reply:
Gotcha. Yeah, I noticed that this PR implements this function at least twice. I think it would make sense to centralize it and improve the name if it's the exact same code.